packages/types/package.json

```diff
@@ -22,7 +22,8 @@
     "test": "echo \"No tests specified\" && exit 0",
     "generate": "yarn generate:models && yarn generate:schemas && yarn build",
     "generate:models": "ts-node generate-models.ts",
-    "generate:schemas": "ts-node generate-schemas.ts"
+    "generate:schemas": "ts-node generate-schemas.ts",
+    "assert-no-changes": "yarn generate:models --failOnChanges"
```

Comment:
Is there a reason we're only running this assertion against the models? Might be good to check if someone has changed any of the schema types and forgotten to update the schemas too

Reply:
Yeah good point, I did consider this and I'm not 100% sure it's needed. Schemas are auto generated from models, so we would catch someone who ran generate:schemas instead of just generate, but that seems a little unlikely. Also the way I was trying to get it going, by doing a git diff was driving me nuts with how difficult it is to develop codeship steps locally 😢 Are you happy for us to go with this for now which should catch almost all cases?

Reply:
> Schemas are auto generated from models

Schemas also come from raw types right? Eg. `@tupaia/types/src/types/request/**` So it's possible someone updated those but never re-ran `generate:schemas`

Instead of a git diff, is there a reason we can't use a similar `if (failOnChanges) read existing file and compare to new text` kind of logic?

All good if I'm missing something here, happy to just have the models validation for now and we can add schema validation if/when it becomes an issue and after we've switched to GHA 👍

Reply:
Yes correct. I was being lazy, added the check on schemas as well.